lib/promscrape/discovery/kuma/api.go

```diff
@@ -0,0 +1,167 @@
+package kuma
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discoveryutils"
+	"github.com/VictoriaMetrics/metrics"
+)
+
+var configMap = discoveryutils.NewConfigMap()
+
+type apiConfig struct {
+	client *discoveryutils.Client
+	path   string
+
+	cancel        context.CancelFunc
```

Comment:
waitGroup is needed to be sure, that all goroutines were correctly stopped and there is no resource leak.

Reply:
the answer is below